include/openssl/ssl.h

```diff
@@ -1558,6 +1558,27 @@ __owur char *SSL_get_srp_username(SSL *s);
 __owur char *SSL_get_srp_userinfo(SSL *s);
 # endif
 
+/*
+ * Early callback and helpers.
+ */
+typedef struct clienthello_msg_st SSL_EARLY;
```

Comment:
Hmmm.....we should perhaps change the name of clienthello_msg_st. That's going to get really confusing.